crates/turborepo-lib/src/client.rs

```diff
@@ -12,14 +12,58 @@ use crate::{get_version, retry::retry_future};
 pub trait UserClient {
     fn set_token(&mut self, token: String);
     async fn get_user(&self) -> Result<UserResponse>;
+    async fn get_teams(&self) -> Result<TeamsResponse>;
+    async fn get_caching_status(&self, team_id: &str) -> Result<CachingStatus>;
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "status")]
+pub enum CachingStatus {
+    #[serde(rename = "disabled")]
+    Disabled,
+    #[serde(rename = "enabled")]
+    Enabled,
+    #[serde(rename = "over_limit")]
+    OverLimit,
+    #[serde(rename = "paused")]
+    Paused,
+}
+
+/// Membership is the relationship between the logged-in user and a particular
+/// team
+#[derive(Debug, Clone, Deserialize)]
+pub struct Membership {
+    role: String,
```

Comment:
The API docs list out the possible values `role` can take: https://vercel.com/docs/rest-api#interfaces/team-limited

I believe it would be safe to limit ourselves to those values via an enum since we're querying a specific version of the api.